docs/scripting-element-object.rst

```diff
@@ -0,0 +1,566 @@
+will be expanded by the specified amount of pixes. And if the value of padding is negative the resulting
+screenshot will be shrunk by the specified amount of pixes.
+
+See more in :ref:`splash-png`.
+
+
+.. _splash-element-jpeg:
+
+element:jpeg
+------------
+
+Return a screenshot of the element in JPEG format
+
+**Signature:** ``ok, shot = element:jpeg{width=nil, scale_method='raster', quality=75, region=nil, pad=0}``
+
+**Parameters:**
+
+* width - optional, width of a screenshot in pixels;
+* scale_method - optional, method to use when resizing the image, ``'raster'``
+  or ``'vector'``;
+* quality - optional, quality of JPEG image, integer in range from ``0`` to ``100``;
+* pad - optional, integer or ``{left, top, right, bottom}`` values of padding
+
+**Returns:** ``ok, shot`` pair. If ``ok`` is nil then error happened during the
+function call; ``shot`` provides an information about error type; otherwise
+``shot`` is a JPEG screenshot data, as a :ref:`binary object <binary-objects>`.
+When the result is empty (e.g. if the element is not visible) ``nil`` is returned.
+
+**Async:** no.
+
+*pad* parameter sets the padding of the resulting image. If it is a single integer then the
+padding from all sides will be equal. If the value of the padding is positive the resulting screenshot
+will be expanded by the specified amount of pixes. And if the value of padding is negative the resulting
+screenshot will be shrunk by the specified amount of pixes.
+
+See more in :ref:`splash-jpeg`.
+
+
+.. _splash-element-visible:
+
+element:visible
+---------------
+
+Check whether the element is visible.
+
+**Signature:** ``visible = element:visible()``
+
+**Returns:** ``visible`` indicated whether
+the element is visible.
+
+**Async:** no.
+
+
+.. _splash-element-text:
+
+element:text
+------------
+
+Fetch a text information from the element
+
+**Signature:** ``text = element:text()``
+
+**Returns:** ``text`` is a text content
+of the element.
+
+**Async:** no.
+
+It tries to return the trimmed value of the following JavaScript ``Node`` properties:
+
+* textContent
+* innerText
+* value
+
+If all of them are empty an empty string is returned.
+
+
+.. _splash-element-info:
+
+element:info
+------------
+
+Get useful information about the element.
+
+**Signature:** ``info = element:info()``
+
+**Returns:** ``info`` is a table with element info.
+
+**Async:** no.
+
+Info is a table with the following fields:
+
+* nodeName - node name in a lower case (e.g. *h1*)
+* attributes - table with attributes names and its values
+* tag - html string representation of the element
+* html - inner html of the element
+* text - inner text of the element
+* x - x coordinate of the element
+* y - y coordinate of the element
+* width - width of the element
+* height - height of the element
+* visible - flag representing if the element is visible
+
+
+.. _splash-element-field-value:
+
+element:field_value
+-------------------
+
+Get value of the field element (input, select).
```

Comment:
it do the following:

if the element is `select` and it has multiples values it returns a table with that values
if the element is `select` and it has single value it returns it
if the element has `type = "radio"` it returns it's value if it's checked and `nil` otherwise
if the element has `type = "checkbox"` it returns bool
otherwise it returns the `value` property value or empty string if it doesn't exist